src/daft-dsl/src/functions/list/get.rs

```diff
@@ -0,0 +1,47 @@
+use crate::Expr;
+use daft_core::{datatypes::Field, schema::Schema, series::Series};
+
+use common_error::{DaftError, DaftResult};
+
+use super::super::FunctionEvaluator;
+
+pub(super) struct GetEvaluator {}
+
+impl FunctionEvaluator for GetEvaluator {
+    fn fn_name(&self) -> &'static str {
+        "get"
+    }
+
+    fn to_field(&self, inputs: &[Expr], schema: &Schema, _: &Expr) -> DaftResult<Field> {
+        match inputs {
+            [input, idx] => {
+                let input_field = input.to_field(schema)?;
+                let idx_field = idx.to_field(schema)?;
+
+                if !idx_field.dtype.is_integer() {
+                    return Err(DaftError::TypeError(format!(
+                        "Expected get index to be integer, received: {}",
+                        idx_field.dtype
+                    )));
+                }
+
+                let exploded_field = input_field.to_exploded_field()?;
```

Comment:
Probably should be running `list_get` here?

Reply:
To my understanding, this part is only for identifying the type of the resulting expression, and execution is only done in `evaluate`. What would be the rationale of running `list_get`?

Reply:
Yes - I misread, you're correct!